_adr/81/index.adoc

```diff
@@ -19,10 +19,264 @@ NOTE: https://datatracker.ietf.org/doc/html/rfc2119[RFC-2119] keywords are used
+`medium`:: Ingress 60MB/s, egress 60MB/s, 2000 partitions, 1000 connections per instance, max 100 connections per second, 1MB max message size.
+`large`:: Ingress 90MB/s, egress 90MB/s, 3000 partitions, 1500 connections per instance, max 100 connections per second, 1MB max message size.
+
+In this example, the functional attributes scale linearly, with the exception of connection rate and max message size.
+In general, linearity SHALL NOT be a requirement, even for those attributes which do scale up.
+For example, it would have been acceptable to define the `large` ingress as 100M/s, even though this is not 3× the `small` ingress, even when the other `large` functional parameters are 3× their `small` values.
+
+=== General principles
+
+* An individual fleet shard is required to provide only a subset of all profiles supported by the service globally. 
+Indeed, a fleet shard might possibly support only a single profile.
+For example a shard on AWS might support a different set of profiles to a shard on Azure, or the same profiles might have different sizes on those clouds.
+One of the purposes of introducing the profile/size concept is to provide flexibility in the kinds of service offered in the future.
+
+* The availability of a profile might be limited to a particular cloud, region or combination of both.
+
+* It is a private concern of the shard exactly how each size in a profile is provisioned.
+In principle it could use vertical and/or horizontal scaling and/or density > 1, in addition to applying quotas to limit capacity to the advertised limits.
+The fleet manager doesn't know.
+
+=== UI/Fleet manager interface
+
+This ADR does not mandate any particular presentation for how the scale options are presented in the UI.
+In particular while there is only one profile defined, there is no requirement to display profiles in the UI at all.
+
+The fleet manager SHALL be the source of truth for the profiles and sizes available, and how those depend on cloud and region. 
+The fleet manager SHALL provide an endpoint which the UI MUST use to determine the available profiles and sizes.
+The UI MUST NOT make assumptions about linearity of scaling, instead it uses only the profiles and sizes provided by the endpoint.
+
+==== Profile discovery
+
+Reusing the "Example" profile defined above, let's illustrate how the Fleet Manager could communicate the available kinds of cluster to a UI.
+
+.Sequence diagram of UI/Fleet manager interaction
+image::ui-fleet_manager.png[UI/Fleet manager interaction]
+
+After the UI has discovered the available cloud providers and regions it would make another GET request `/api/kafkas_mgmt/v1/profile/{cloudId}/{cloudRegion}` to discover the profiles available in this cloud and region.
+An example response JSON is shown below.
+
+[source,json]
+.Example profile available to a UI from the Fleet Manager's `GET /api/kafkas_mgmt/v1/profiles/{cloudId}/{cloudRegion}` endpoint
+----
+"profiles": [ <1>
```

Comment:
@pmuir do you think this format offers advantages for the UI (metadata) or do you think the current format described in https://docs.google.com/document/d/1VeZae6svCNVJS4XbeXL1q6EKdUh7Oe9qmJ4CDGV6rCA/edit#heading=h.m1mukm7lqaex are sufficient?

Reply:
The UI can handle either I think. For me it's more about which API we think is more understandable and better represents what we are doing. I'd argue this one does (except perhaps the name and desription fields are not useful as we want to localize them).

Reply:
Is the top level `parameters` field useful? It defines names for all those parameters but I assume you will localization them too on the UI?

Reply:
The only difference then is that sizes are a subarray. We could represent them in app-interface like this:

```yaml
    SUPPORTED_KAFKA_SIZES:
    - id: eval
      sizes: 
      -  id: x1
         ingressThroughputPerSec: “30Mi”
         egressThroughputPerSec: “30Mi”
         totalMaxConnections: 3000
         maxDataRetentionSize: “1000Gi”
         maxPartitions: 1000
         maxDataRetentionPeriod: “P14D”
         maxConnectionAttemptsPerSec: 100
    - id: standard
      sizes:
      -  id: x1
         ingressThroughputPerSec: “30Mi”
         egressThroughputPerSec: “30Mi”
         totalMaxConnections: 3000
         maxDataRetentionSize: “1000Gi”
         maxPartitions: 1000
         maxDataRetentionPeriod: “P14D”
         maxConnectionAttemptsPerSec: 100
```

Reply:
Yes, the UI can localize these based on the id. My comment was just whether the name and description fields were therefore useful (as they won't be used).

Reply:
Ah, OK. Out of interest, is there consistent l10n in the CLI @wtrocki ?

Reply:
Yes.

Reply:
I've tweaked the ADR, removing the `name` and `description` fields. I've also added "cost" as a parameter @pb82 pointed out that that was needed at least by the fleet manager so it knows about the AMQ quota cost. I'm thinking this is something that the UI also perhaps needs to surface, and therefore it should be in this API.